lib/runner.js

```diff
@@ -1027,11 +1028,9 @@ Runner.prototype.run = function(fn, opts) {
     debug(constants.EVENT_RUN_END);
     debug('run(): emitted %s', constants.EVENT_RUN_END);
     fn(self.failures);
+    self._removeEventListener(process, 'uncaughtException', self.uncaught);
```

Comment:
Removing the uncaught exception handler here is incorrect, too early, IMO. There can be exceptions bubbling up after runner's end. We even have a check for this (`STATE_STOPPED`) in `Runner#_uncaught`.

Reply:
If a runner `'end'`s, then results are likely to have been reported based on that runner. If a test could error post-end, it's too late to include it in results... Simultaneously, if a test still has outstanding actions like that, the runner probably shouldn't be `'end'`ing!